src/v/cloud_storage/topic_manifest.cc

```diff
@@ -41,68 +193,88 @@ ss::future<> topic_manifest::update(ss::input_stream<char> is) {
-    }
-    auto ns = model::ns(m["namespace"].GetString());
-    auto tp = model::topic(m["topic"].GetString());
-    int32_t partitions = m["partition_count"].GetInt();
-    int16_t rf = m["replication_factor"].GetInt();
-    int32_t rev = m["revision_id"].GetInt();
-    cluster::topic_configuration conf(ns, tp, partitions, rf);
-    // optional
-    if (!m["compression"].IsNull()) {
-        conf.properties.compression = boost::lexical_cast<model::compression>(
-          m["compression"].GetString());
-    }
-    if (!m["cleanup_policy_bitflags"].IsNull()) {
-        conf.properties.cleanup_policy_bitflags
-          = boost::lexical_cast<model::cleanup_policy_bitflags>(
-            m["cleanup_policy_bitflags"].GetString());
-    }
-    if (!m["compaction_strategy"].IsNull()) {
-        conf.properties.compaction_strategy
-          = boost::lexical_cast<model::compaction_strategy>(
-            m["compaction_strategy"].GetString());
-    }
-    if (!m["timestamp_type"].IsNull()) {
-        conf.properties.timestamp_type
-          = boost::lexical_cast<model::timestamp_type>(
-            m["timestamp_type"].GetString());
-    }
-    if (!m["segment_size"].IsNull()) {
-        conf.properties.segment_size = m["segment_size"].GetInt64();
-    }
-    // tristate
-    if (m.HasMember("retention_bytes")) {
-        if (!m["retention_bytes"].IsNull()) {
-            conf.properties.retention_bytes = tristate<size_t>(
-              m["retention_bytes"].GetInt64());
-        } else {
-            conf.properties.retention_bytes = tristate<size_t>(std::nullopt);
+    rapidjson::IStreamWrapper wrapper(stream);
+    rapidjson::Reader reader;
+    topic_manifest_handler handler;
+    if (reader.Parse(wrapper, handler)) {
+        vlog(cst_log.debug, "Parsed successfully!");
+        if (handler._version != topic_manifest_version) {
+            throw std::runtime_error(fmt_with_ctx(
+              ssx::sformat,
+              "topic manifest version {} is not supported",
+              handler._version));
         }
-    }
-    if (m.HasMember("retention_duration")) {
-        if (!m["retention_duration"].IsNull()) {
-            conf.properties.retention_duration
-              = tristate<std::chrono::milliseconds>(
-                std::chrono::milliseconds(m["retention_duration"].GetInt64()));
+        _rev = handler._revision_id.value();
+        _topic_config = handler._topic_config;
+        if (handler.compaction_strategy_sv) {
+            try {
+                _topic_config->properties.compaction_strategy
+                  = boost::lexical_cast<model::compaction_strategy>(
+                    handler.compaction_strategy_sv.value());
+            } catch (const std::runtime_error& e) {
+                throw std::runtime_error(fmt_with_ctx(
+                  fmt::format,
+                  "Wrong compaction_strategy value: {}",
+                  handler.compaction_strategy_sv.value()));
```

Comment:
I think a lot of these errors can have more context, perhaps of the form:
```c++
            throw std::runtime_error(fmt_with_ctx(
              fmt::format,
              "Failed to parse topic manifest {}: Invalid compaction_strategy {}",
              get_manifest_path(),
              handler.compaction_strategy_sv.value());
```

Reply:
`get_manifest_path()` generates manifest path based on parsed `_topic_config`, so if we failed to parse, then we don't have manifest path